github/github_test.go

```diff
@@ -538,6 +538,42 @@ func TestCheckResponse(t *testing.T) {
 	}
 }
 
+func TestCheckResponse_withBlock(t *testing.T) {
+	res := &http.Response{
+		Request:    &http.Request{},
+		StatusCode: http.StatusBadRequest,
+		Body: ioutil.NopCloser(strings.NewReader(`{"message":"m",
+			"errors": [{"resource": "r", "field": "f", "code": "c"}],
+			"block": {"reason": "dmca", "created_at": "2016-03-17T15:39:46+00:00"}}`)),
+	}
+	err := CheckResponse(res).(*ErrorResponse)
+
+	if err == nil {
+		t.Errorf("Expected error response.")
+	}
+
+	want := &ErrorResponse{
+		Response: res,
+		Message:  "m",
+		Errors:   []Error{{Resource: "r", Field: "f", Code: "c"}},
+		Block: &Block{
+			Reason:    "dmca",
+			CreatedAt: &Timestamp{time.Date(2016, 3, 17, 15, 39, 46, 0, time.UTC)},
+		},
+	}
+	if err.Block == nil && err.Block.CreatedAt == nil {
+		t.Errorf("CreatedAt = nil, want %v", want.Block.CreatedAt)
+	} else if !want.Block.CreatedAt.Equal(*err.Block.CreatedAt) {
+		t.Errorf("CreatedAt = %v, want %v", err.Block.CreatedAt, want.Block.CreatedAt)
+	}
+	// Clear times for DeepEqual to succeed.
+	err.Block.CreatedAt = nil
```

Comment:
I'm surprised this is necessary.  DeepEqual fails even for time values that are identical?